src/scheduler/job_info.c

```diff
@@ -3037,6 +3057,13 @@ find_and_preempt_jobs(status *policy, int pbs_sd, resource_resv *hjob, server_in
 	if (hjob->aoename != NULL)
 		return 0;
 
+	/* If a job couldn't run because of max_run_subjobs limit, do not try preemption */
+	if (hjob->job->is_array && err->error_code == MAX_RUN_SUBJOBS) {
+		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, hjob->name,
+			  "Job cannot run due to max_run_subjobs limit, preemption not attempted");
+		return 0;
+	}
```

Comment:
I'll add it there because that seems to be the right place for it. But the reason I added it up top was that there isn't really a need to go forward and call is_ok_to_run again. 
In the case of qrun request, the error structure main_sched_loop passes to this function will not have MAX_RUN_SUBJOBS error set in it because is_ok_to_run would not set it and it runs the job. Same will happen with the second call to is_ok_to_run in this function.